service/storage_service.cc

```diff
@@ -669,7 +670,14 @@ future<> storage_service::topology_state_load() {
         // since it is not loaded in join_cluster in the
         // raft_topology_change_enabled() case.
         if (!_gossiper.get_endpoint_state_ptr(ep)) {
-            co_await _gossiper.add_saved_endpoint(ep, permit.id());
+            gms::loaded_endpoint_state st;
+            const auto& topo = tmptr->get_topology();
+            // node must exist in topology if it's in tmptr->get_all_endpoints
+            const auto* node = topo.find_node(e);
```

Comment:
why?  dereferencing a null pointer, that should never happen, would result in a crash too.
anyhow I can use `get_node` that was introduced recently, and it would throw a runtime_error is the node was not found.

Reply:
IMO on_internal_error is better than regular runtime_error --- so we get an easier to debug failure in CI if it happens

Reply:
done in v10